NativeUI/Elements/NativeText.cs

```diff
@@ -0,0 +1,193 @@
+using GTA.Native;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+
+namespace NativeUI.Elements
+{
+    /// <summary>
+    /// The alignment of the text to draw.
+    /// </summary>
+    public enum TextAlignment
+    {
+        Left,
+        Centered,
+        Right,
+    }
+
+    /// <summary>
+    /// New Design for the NativeUI texts.
+    /// </summary>
+    public class NativeText : NativeElement
+    {
+        #region Consistent Values
+
+        /// <summary>
+        /// Max size of every Caption chunk.
+        /// </summary>
+        private const int ChunkSize = 90;
+
+        #endregion
+
+        #region Private Properties
+
+        /// <summary>
+        /// The raw word wrap value.
+        /// </summary>
+        private float InternalWordWrap = 0f;
+        /// <summary>
+        /// The already calculated word wrapping.
+        /// </summary>
+        private float CalculatedWrap = 0f;
+        /// <summary>
+        /// Internal width calculation.
+        /// </summary>
+        private float Width = 0f;
+        /// <summary>
+        /// List of chunks of the Caption.
+        /// </summary>
+        private List<string> Chunks = new List<string>();
+        /// <summary>
+        /// Internal storage of the caption.
+        /// </summary>
+        private string _Caption = "";
+
+        #endregion
+
+        #region Public Properties
+
+        /// <summary>
+        /// The UTF-8 string to draw on the screen.
+        /// </summary>
+        public string Caption
+        {
+            get => _Caption;
+            set // F
+            {
+                _Caption = value;
+                Chunks.Clear(); // Wipe the list
+                double TotalParts = Math.Ceiling(value.Length / (float)ChunkSize); // Get the number of parts that we need
```

Comment:
This is wrong. You have to compare UTF8 byte count, not character count. Here's a string to test it out:

`АБВГДЕЁЖЗИЙКЛМНОПРСТУФХЦЧШЩЪУЬЭЮЯАБВГДЕЁЖЗИЙКЛМНОПРСТУФХЦЧШЩЪУЬЭЮЯ`

Character count: 66
Byte count: 132

With the current code it cuts out at `АБВГДЕЁЖЗИЙКЛМНОПРСТУФХЦЧШЩЪУЬЭЮЯАБВГДЕЁЖЗИЙКЛМНО`

Reply:
Here's the fixed code. It looks like splitting UTF8 characters is fine when pushing them. 

```
_Caption = value;
Chunks.Clear(); // Wipe the list
int ByteCount = System.Text.Encoding.UTF8.GetByteCount(value);
int TotalParts = (int) Math.Ceiling(ByteCount / (float)ChunkSize); // Get the number of parts that we need
if (TotalParts == 1) // If we only need one part
{
    Chunks.Add(value); // Add the complete string
}
else // Otherwise
{
    byte[] bytes = System.Text.Encoding.UTF8.GetBytes(value);
    for (int i = 0; i < TotalParts; i++) // Iterate over the number of parts that we need
    {
        int Start = i * ChunkSize; // Calculate the start
        // If the Start + Aditional Chunk is higher than the length of the string, add the characters left
        int Count = Math.Min(ChunkSize, ByteCount - Start);
        Chunks.Add(System.Text.Encoding.UTF8.GetString(bytes, Start, Count)); // Finally add the string
    }
}
```

Reply:
Next time use the option to recommend code please.

Reply:
Anyway, is done on 6b716fe2abc3ffd16e5247d90f4ca223f0b09ac5.

Reply:
Unfortunately suggestions are one line only as far as I'm aware